lib/AST/Decl.cpp

```diff
@@ -5062,6 +5072,45 @@ bool FuncDecl::isDeferBody() const {
   return getName() == getASTContext().getIdentifier("$defer");
 }
 
+bool FuncDecl::isXCTestUnitTest() const {
+  if (!hasName())
+    return false;
+
+  // A 'test candidate' is:
+  // 1. An instance method...
+  if (!isInstanceMember())
+    return false;
+
+  // 2. ...on a class or extension (not a struct) subclass of XCTestCase...
+  auto parentClass = getDeclContext()->getAsClassOrClassExtensionContext();
+  if (!parentClass)
+    return false;
+  if (!parentClass->isXCTestCaseSubclass())
+    return false;
+
+  // 3. ...that returns void...
+  Type retTy = getResultInterfaceType();
+  if (retTy && !retTy->isVoid())
+    return false;
+
+  // 4. ...takes no parameters...
```

Comment:
Do we really want to hardcode so many details? For IBAction it definitely bit us.